### ISSUE TITLE
Node Name

### ISSUE THREAD

**Issue by ChewyChop** [loganmarchione/hassos-addons#21]
Issue with having multiple home assistant servers in multiple locations, the node name reports 0d869efa-prometheus-node-exporter for all node names.

**Comment by loganmarchione** [comment]
Hmmm, never thought about that. Mine is the same name as yours, so it must be the hash of the container or something? I'll have to see if I can change that. Not sure if you can set a hostname inside a container...

In the meantime, try changing the `label` on your Prometheus server. Here is a sample from my `prometheus.yml` file...

```
  - job_name: 'homeassistant'
    static_configs:
      - targets: ['hass02.internal.mydomain.com:9100']
        labels:
          nodename: 'homeassistant'
```

**Comment by ChewyChop** [comment]
Got it!  Thanks!

**Comment by sjthespian** [comment]
Any more thoughts on this one? I'm trying to have a single scraper config for all of my hosts, so overriding it on the prometheus side of things is problematic. I've been digging around the node_exporter docs, and I don't see anything obvious that will do it there...

**Comment by loganmarchione** [comment]
I haven't found a way to override the hostname via the container 🤷🏻‍♂️

**Comment by mateuszdrab** [comment]
What about something like mounting the `/etc/hostname` file into the container?

**Comment by loganmarchione** [comment]
Still no updates here. I posted in the HA discord asking for help. Everything I do to `/etc/hostname` does nothing for node exporter. Right now the best way is to still override it on the Prometheus server side.

**Comment by mateuszdrab** [comment]
Are you bind mounting the host file system in a sub path in the container and then setting the appropriate flags when starting node exporter?

https://github.com/prometheus/node_exporter/issues/1237

I never was affected by this issue because I do manually relabel the metrics but I can see the node name label is wrong indeed.

**Comment by arsenicks** [comment]
Hey, I went down the rabbit hole because I didn't like the fact that in my node exporter full dashboard was showing "0d869efa-prometheus-node-exporter" as a nodename for my home assistant. So for people finding this post, there's a workaround described above and I might have found a fix but I don't have all the addon dev setup available to someone else will have to test that out.

### Workaround:
Replace the label in prometheus, quick, easy fix but still just a workaround.
```
  - job_name: 'node'
    static_configs:
      - targets: ['192.168.x.xxx:9100']
        labels:
          nodename: 'homeassistant'
```


### Potential fix 1:
I think the correct way to do this would simply to show the docker host hostname.

From what I saw HA base container used by this addon are using alpine:3.22
Here's the output of "hostname -f"(should return full hostname.domainname) on a "normal" alpine container:

```
arsenick@nwmsrv:~/docker$ docker run -it alpine:3.22.1 hostname -f
8539f74abb11
arsenick@newmediaserver:~/media-docker$ docker run --name prettyhostname -it alpine:3.22.1 hostname -f
506571430d15
```
So this is the container id of the container itself, not the docker host hostname or even the name of the container.

On a normal docker host, using "--net host" does make the alpine container use the hostname of the docker host:

```
arsenick@nwmsrv:~/docker$ docker run --net host -it alpine:3.22.1 hostname -f
nwmsrv
```
Same result using your addon container on a standard docker host..
```
arsenick@nwmsrv:~/docker$ docker run --net host -it ghcr.io/loganmarchione/hassos-addons/prometheusnodeexporter-amd64:1.0.5 hostname -f
nwmsrv
```
Sooo I'm writing this because it might give other people idea on where to go from here but from what I can see this method might never work because for security reasons and to avoid conflict it seems to set the addon(container) hostname and used the "addon id" and the name of the addon so in our case: "0d869efa-prometheus-node-exporter".


### Potential fix 2:
I can't test this myself, if someone want to try it, it's to set the "host_uts: true" in the addon config.yaml. It's in the official documentation https://developers.home-assistant.io/docs/add-ons/configuration/#optional-configuration-options

Even if this work I'm pretty sure this would be a bad idea or too risky to use..

### Potential fix 3:
That might be the best/easiest solution, but again need someone to test it. Expose a configurable option in config.yaml for the nodename and set the hostname in the startup script of the container.
Ex. Add this to [config.json](https://github.com/loganmarchione/hassos-addons/blob/main/prometheus_node_exporter/config.json)
options:
  nodename: "myhost"
schema:
  nodename: str

and then we add something along those line to [run.sh](https://github.com/loganmarchione/hassos-addons/blob/main/prometheus_node_exporter/rootfs/run.sh)
`hostname "$(bashio::config 'hostname')"`

From what I can see that would change what uname -n return which I guess is the same returned by the uname system call made by "node_uname_info".


So TL;DR fix the nodename using the label is the easiest way for now due to supervisor security and limitation. Maybe we could simply update the README to clarify this and explain how to do it while we search and test the proposed solution in this post ?

**Comment by loganmarchione** [comment]
Hey everyone, thanks for making suggestions for this issue.

Looking at the [docs](https://developers.home-assistant.io/docs/add-ons/communication/#network), it says

>Names/aliases are used for communication inside Home Assistant.
The name is generated using the following format: `{REPO}_{SLUG}`, e.g., `local_xy` or `3283fh_myaddon`. In this example, `{SLUG}` is defined in an add-on's `config.yaml` file. You can use this name as the DNS name also, but you need to replace any `_` with `-` to have a valid hostname. If an add-on is installed locally, `{REPO}` will be `local`. If the add-on is installed from a Github repository, `{REPO}` is a hashed identifier generated from the GitHub repository's URL (ex: `https://github.com/xy/my_hassio_addons`). See [here](https://github.com/home-assistant/supervisor/blob/4ac7f7dcf08abb6ae5a018536e57d078ace046c8/supervisor/store/utils.py#L17) to understand how this identifier is generated. Note that this identifier is required in certain actions that use the [Supervisor add-on API][supervisor-addon-api]. You can view the repository identifiers for all currently installed add-ons via a GET request to the Supervisor API `addons` endpoint.

So, `REPO` is the first 8 characters of the SHA1 hash of my repo:

```
echo -n "https://github.com/loganmarchione/hassos-addons" | sha1sum | head -c 8
0d869efa
```

And `SLUG` is `prometheus_node_exporter`, but with dashes instead of underscores.

This results in the string `0d869efa-prometheus-node-exporter` that everyone is seeing.

I'm running Home Assistant Operating System, so I don't have access to the Docker host or the ability to run `docker` commands on the host. In addition, this container is already running on the host's network.

I also don't know if changing the hostname would do anything, since Supervisor controls the container lifecycle, and would reset it to REPO_SLUG on each restart. The [API calls](https://developers.home-assistant.io/docs/api/supervisor/endpoints/#add-ons) that Supervisor uses would also break, right?

```
curl -sSL -H "Authorization: Bearer $SUPERVISOR_TOKEN" http://supervisor/addons/0d869efa_prometheus_node_exporter/info | jq
```

I did try to manually set the hostname...

```
#!/usr/bin/env bashio
# ==============================================================================
# Home Assistant Community Add-on: Prometheus Node Exporter
# Runs the Prometheus Node Exporter
# ==============================================================================
bashio::log.info "Starting Prometheus Node Exporter..."

bashio::log.info "Get the actual HA system hostname..."
REAL_HOSTNAME=$(bashio::host.hostname)

# Change the container's hostname
bashio::log.info "Set container hostname to: ${REAL_HOSTNAME}"
hostname "${REAL_HOSTNAME}" 2>/dev/null || true

# Run Prometheus Node Exporter
exec /usr/local/bin/node_exporter \
  --web.config.file=/config/prometheus_node_exporter/node_exporter_web.yml \
  $(bashio::config 'cmdline_extra_args' '')
```

That results in this in the logs...
```
[13:25:25] INFO: Starting Prometheus Node Exporter...
[13:25:25] INFO: Get the actual HA system hostname...
[13:25:25] INFO: Set container hostname to: hass02
```

But, the scrape config still shows `local-prometheus-node-exporter` (`local` in this case because I was testing with a local add-on).

```
node_uname_info{domainname="local.hass.io",machine="x86_64",nodename="local-prometheus-node-exporter",release="6.12.43-haos",sysname="Linux",version="#1 SMP PREEMPT_DYNAMIC Mon Sep  8 17:03:23 UTC 2025"} 1
```

I'm really unsure if there is a fix for this, other than adding `labels` in the scrape config. I'm going to close this issue and update the README to add a FAQ about this.
